release-tools/scripts/docker_download_plugins.sh

```diff
@@ -14,32 +16,40 @@ S3_RELEASE_BASEURL=`yq eval '.urls.ODFE.releases' $MANIFEST_FILE`
 S3_RELEASE_FINAL_BUILD=`yq eval '.urls.ODFE.releases_final_build' $MANIFEST_FILE | sed 's/\///g'`
 S3_RELEASE_BUCKET=`echo $S3_RELEASE_BASEURL | awk -F '/' '{print $3}'`
 
-PLUGIN_DIR="$REPO_ROOT/elasticsearch/docker/build/elasticsearch/plugins"
+PLUGIN_DIR="$REPO_ROOT/$CATEGORY/docker/build/$CATEGORY/plugins"
 PLUGIN_PATH=`yq eval '.urls.ODFE.releases' $MANIFEST_FILE | sed "s/^.*$S3_RELEASE_BUCKET\///g"`
 
 if [ -z "$S3_RELEASE_FINAL_BUILD" ]
 then
   S3_RELEASE_BUILD=`aws s3api list-objects --bucket $S3_RELEASE_BUCKET --prefix "${PLUGIN_PATH}${OD_VERSION}" --query 'Contents[].[Key]' --output text | awk -F '/' '{print $3}' | uniq | tail -n 1`
   echo Latest: $S3_RELEASE_BUILD
 else
+  S3_RELEASE_BUILD=$S3_RELEASE_FINAL_BUILD
   echo Final: $S3_RELEASE_BUILD
 fi
 
 
 # Please DO NOT change the orders, they have dependencies
-PLUGINS_ARRAY=( `$REPO_ROOT/release-tools/scripts/plugins-info.sh elasticsearch-plugins plugin_basename` )
+PLUGINS_ARRAY=( `${REPO_ROOT}/release-tools/scripts/plugins-info.sh ${CATEGORY}-plugins plugin_basename` )
 
 rm -rf $PLUGIN_DIR
 mkdir -p $PLUGIN_DIR
 
 for index in ${!PLUGINS_ARRAY[@]}
 do
-  plugin_latest=`aws s3api list-objects --bucket $S3_RELEASE_BUCKET --prefix "${PLUGIN_PATH}${OD_VERSION}/$S3_RELEASE_BUILD/elasticsearch-plugins" --query 'Contents[].[Key]' --output text | grep -v sha512 | grep ${PLUGINS_ARRAY[$index]} | grep zip | sort | tail -n 1`
+  if echo ${PLUGINS_ARRAY[$index]} | grep -i reports
+  then
+    plugin_latest=`aws s3api list-objects --bucket $S3_RELEASE_BUCKET --prefix "${PLUGIN_PATH}${OD_VERSION}/${S3_RELEASE_BUILD}/${CATEGORY}-plugins" --query 'Contents[].[Key]' --output text \
+                   | grep -v sha512 | grep ${PLUGINS_ARRAY[$index]} | grep zip | grep linux | grep "$ARCHITECTURE" | sort | tail -n 1`
+  else
+    plugin_latest=`aws s3api list-objects --bucket $S3_RELEASE_BUCKET --prefix "${PLUGIN_PATH}${OD_VERSION}/${S3_RELEASE_BUILD}/${CATEGORY}-plugins" --query 'Contents[].[Key]' --output text \
+                   | grep -v sha512 | grep ${PLUGINS_ARRAY[$index]} | grep zip | sort | tail -n 1`
+  fi
 
   if [ ! -z "$plugin_latest" ]
```

Comment:
We will need another condition to confirm it is a release candidate.

Reply:
Like we talked, it is already checked in https://github.com/opendistro-for-elasticsearch/opendistro-build/blob/master/release-tools/scripts/staging-copy-artifacts.py#L155